[tests] Flaky test test_devicegroup_list_api

```
FAIL: test_devicegroup_list_api (openwisp2.sample_config.tests.TestConfigApi) [should not contain default or required templates]
  (0.53s) test_ip_not_in_add_device
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/openwisp-controller/openwisp-controller/openwisp_controller/config/tests/test_api.py", line 687, in test_devicegroup_list_api
    self.assertNotContains(r, t2.name)
  File "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/django/test/testcases.py", line 481, in assertNotContains
    self.assertEqual(real_count, 0, msg_prefix + "Response should not contain %s" % text_repr)
AssertionError: 2 != 0 : Response should not contain 't2'
```

--- a/openwisp_controller/config/tests/test_api.py
+++ b/openwisp_controller/config/tests/test_api.py
@@ -683,8 +683,12 @@ class TestConfigApi(
             t1 = self._create_template(name='t1')
             t2 = self._create_template(name='t2', required=True)
             r = self.client.get(path, {'format': 'api'})
-            self.assertContains(r, t1.name)
-            self.assertNotContains(r, t2.name)
+            self.assertContains(
+                r, f'<option value="{t1.id}">{t1.name}</option>', html=True
+            )
+            self.assertNotContains(
+                r, f'<option value="{t2.id}">{t2.name}</option>', html=True
+            )
 
     def test_devicegroup_detail_api(self):
         device_group = self._create_device_group()

